src/cephadm/cephadmlib/daemon_identity.py

```diff
@@ -17,6 +17,9 @@ def __init__(
         self._fsid = fsid
         self._daemon_type = daemon_type
         self._daemon_id = str(daemon_id)
+        assert self._fsid
+        assert self._daemon_type
+        assert self._daemon_id
```

Comment:
Can I ask for the purpose of this? The types aren't marked optional so from a mypy POV it should already be making sure they're actually set and when actually running for a real cluster I wouldn't think just raising an assertion error would do us any good. Is this one of those commits to just check if we would ever hit this case when we do a teuthology run or something?